packages/@aws-cdk/cdk/lib/util/physical-name-generator.ts

```diff
@@ -0,0 +1,58 @@
+import crypto = require('crypto');
+import { Stack } from '../cloudformation/stack';
+import { IConstruct } from '../core/construct';
+
+export class PhysicalNameGenerator {
+  public static from(construct: IConstruct): PhysicalNameGenerator {
```

Comment:
What's wrong with `construct.node.uniqueId`? I believe it includes all this information (and more). Using `node.id` is not going to be sufficient since it's not globally unique within an `App` (but `uniqueId` is)

Reply:
You are correct, it should be `uniqueId` - this was a mistake on my part. Will change.

Reply:
Damn, missed this one as well. Will change in the next revision.

Reply:
Fixed.